### PR TITLE
Constexpr abs/fabs

### PR DESCRIPTION
**Pull request by mborland**

### CODE REVIEW

**Comment by mborland** [comment]
@jzmaddock and @NAThompson I think this is ready for review. The drone build failures are all network timeout issues.

File: doc/sf/ccmath.qbk [hunk 1]
```diff
@@ -9,7 +9,10 @@ LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 [heading Description]
 
-Constexpr implementations of the functionality found in `<cmath>`. The following functions require C++17 or greater.
+Constexpr implementations of the functionality found in `<cmath>`.
+In a constexpr context the functions will use an implementation defined in boost.
```
Comment (NAThompson) [inline doc/sf/ccmath.qbk:13]: Could you make sure `constexpr` is formatted as code?

File: test/ccmath_abs_test.cpp [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+#include <boost/multiprecision/float128.hpp>
+#endif
+
+template <typename T>
+void test()
+{
+    static_assert(boost::math::ccmath::abs(T(3)) == 3);
+    static_assert(boost::math::ccmath::abs(T(-3)) == 3);
+    static_assert(boost::math::ccmath::abs(T(-0)) == 0);
+    static_assert(boost::math::ccmath::abs(-std::numeric_limits<T>::infinity()) == std::numeric_limits<T>::infinity());
```
Comment (NAThompson) [inline test/ccmath_abs_test.cpp:23]: Could some very simple expressions be joined here? eg

```
static_assert(abs(sin(1)) == ...)
```

Comment (mborland) [reply test/ccmath_abs_test.cpp:23]: The difficulty is the only functions guaranteed to be `constexpr` are `pow` and `sqrt`. I could do something heavily restricted like only testing your suggestion on GCC10+ where some of `cmath` is `constexpr`. Thoughts?

Comment (NAThompson) [reply test/ccmath_abs_test.cpp:23]: Yeah, I think if you have a guarantee like that, then you should *try* to test it. If it dies in CI, well, not a big deal.

**Comment by NAThompson** [comment]
Unrelated point: I feel like after this we should consider doing another standalone release.

File: test/ccmath_abs_test.cpp [hunk 3]
```diff
@@ -28,6 +29,18 @@ void test()
...
+template <typename T>
+void gpp_test()
+{
+    constexpr T tol = std::numeric_limits<T>::epsilon();
+
+    static_assert(std::sin(T(0)) == 0);
+    
+    constexpr T sin_1 = boost::math::ccmath::abs(std::sin(T(-1)));
+    static_assert(sin_1 > 0);
+    static_assert(boost::math::ccmath::abs(sin_1 - 0.8414709848078965066525l) < tol);
```
Comment (NAThompson) [inline test/ccmath_abs_test.cpp:41]: I believe this should be bitwise equal to the correctly rounded result, so `abs(sin_1 - expected) < tol` is too loose. Maybe use a hexadecimal floating point literal and check for bitwise equality?

Comment (NAThompson) [reply test/ccmath_abs_test.cpp:41]: Nice test, btw.

**Comment by mborland** [comment]
@NAThompson All of the drone build failures are timeouts

**Comment by NAThompson** [comment]
I'm happy with these changes, but this is something @jzmaddock understands better than me.

Also make sure to squash merge since there were a bunch of "fight the CI commits" which will make `git bisect` painful.

**Pull request merged**